src/transform/ancestor.js

```diff
@@ -118,7 +118,7 @@ function canBeLifted(doc, range) {
   }
 }
 
-// :: (Node, Pos, ?Pos) → bool
+// :: (Node, Pos, ?Pos) → ?{found: any, range: any}
```

Comment:
Hm, that also sounds like basic control flow analysis though -- if there's no terminating condition and no `break`, then control won't exit the bottom of the loop. (Of course, it's possible that I have loops with bogus terminating conditions -- if so, let me know.)